### PR TITLE
L2-372: Merge Maturity

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

Add method to merge maturity of a neuron.

# Changes

* New method "mergeMaturity" on Governance Service.

# Tests

* Test new method.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker @mstrasinskis please review

I wasn't able to test with a neuron that has maturity. Since I don't have one locally with enough maturity.

Yet, when I tried it with a neuron without maturity, I got the error that the user can't merge maturity if it's less than the transaction fee. Which we can assume it's the right error.

To properly test maturity features, we need to prepare a testnet with special governance canister or run the replica locally with a special governance canister. @bitdivine is working on it.

File: src/canisters/governance/request.converters.ts [hunk 1]
```diff
@@ -867,6 +867,22 @@ export const toDisburseNeuronRequest = ({
     },
   });
 
+export const toMergeMaturityRequest = ({
+  neuronId,
+  percentageToMerge,
+}: {
+  neuronId: NeuronId;
+  percentageToMerge: number;
```
Comment (peterpeterparker) [inline src/canisters/governance/request.converters.ts:875]: do we need a check that test the percentage is between 0 and 100?

**Review by peterpeterparker — COMMENTED** [review]
Code looks good, just one question

File: src/canisters/governance/request.converters.ts (hunk 1 above)
Comment (lmuntanerDfinity) [reply src/canisters/governance/request.converters.ts:875]: Good question, I guess it makes sense. The error is done in the backend, but we can also do it here.

Comment (peterpeterparker) [reply src/canisters/governance/request.converters.ts:875]: Good call. We defer other rules to the backend too.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

File: src/canisters/governance/request.converters.ts (hunk 1 above)
Comment (lmuntanerDfinity) [reply src/canisters/governance/request.converters.ts:875]: Check added.

Comment (peterpeterparker) [reply src/canisters/governance/request.converters.ts:875]: Coolio

**Pull request merged**